devito/finite_differences/differentiable.py

```diff
@@ -145,24 +186,48 @@ def __eq__(self, other):
         return super(Differentiable, self).__eq__(other) and\
             all(getattr(self, i, None) == getattr(other, i, None) for i in self._state)
 
+    def index(self, dim):
+        for d in self.dimensions:
+            if d is dim:
+                return d
+        return None
+
+    @property
+    def name(self):
+        return "".join(f.name for f in self._functions)
+
     @property
     def laplace(self):
         """
         Generates a symbolic expression for the Laplacian, the second
         derivative w.r.t all spatial Dimensions.
         """
-        space_dims = [d for d in self.indices if d.is_Space]
+        space_dims = [d for d in self.dimensions if d.is_Space]
         derivs = tuple('d%s2' % d.name for d in space_dims)
         return Add(*[getattr(self, d) for d in derivs])
 
+    @property
+    def div(self):
+        space_dims = [d for d in self.dimensions if d.is_Space]
+        derivs = tuple('d%s' % d.name for d in space_dims)
+        return Add(*[getattr(self, d) for d in derivs])
+
+    @property
+    def grad(self):
+        from devito.types.tensor import VectorFunction, VectorTimeFunction
+        comps = [getattr(self, 'd%s' % d.name) for d in self.dimensions if d.is_Space]
+        vec_func = VectorTimeFunction if self.is_TimeDependent else VectorFunction
+        return vec_func(name='grad_%s' % self.name, time_order=self.time_order,
+                        space_order=self.space_order, components=comps, grid=self.grid)
+
     def laplace2(self, weight=1):
```

Comment:
I prefer calling this the 'biharmonic' operator but won't loose any sleep if you prefer not to change.

Reply:
Sure, maybe for a follow up cleanup PR